SelfDriveEnv.py

```diff
@@ -270,79 +154,394 @@ def handle_events(self):
                 if event.key == pygame.K_ESCAPE:
                     self.close_window()
 
-    def reset(self, new=False):
-        if new:
-            self.open_window()
-            while self.screen:
-                self.render()
-            for row in self.track:
-                for border in row:
-                    border.mutable = False
-            self.save_track()
-        else:
-            self.load_track()
-        self.cars[0].reset()
-        self.start_locs = self.calc_avg_pos("start")
-        self.finish_locs = self.calc_avg_pos("finish")
-        self.cars[0].set_pos(self.start_locs)
-        self.initialized = True            
-        return self.cars[0].update_sensors()
-
-        
     def colliding_with(self, x, y):    
         for row in self.track:
             for border in row:
-                if border.rect.collidepoint(x,y) and border.active and \
-                    not border.start_finish:
-                    return border.index
-        return False
+                if border.rect.collidepoint(x,y) and border.active:                    
+                    return True, border.start_finish, border.index
+                    
+        return False, None, None
```

Comment:
Wow, I liked how u changed how the "start_finish" is actually returned so that it could processed separately in other functions instead of just killing off the information within the "colliding_with()". Seemed like a lot of work adding it to every function that had something to do with colliding, but I agree that it adds more clarity.

Reply:
:))